### PR TITLE
Test E2E that's check both valid and fail proofs should pay

### PR DESCRIPTION
**Pull request by la10736**
1. Tests are listed by search in the folder
2. A new E2E that checks both valid and fail proof pay.

### CODE REVIEW

**Review by drgora — APPROVED** [review]
I believe we should refactor e2e tests to create a `util` library to avoid code duplication, but we can do this in a future PR.

LGTM, thanks!

**Review by ptagl — APPROVED** [review]

**Review by a-petrini — COMMENTED** [review]
Looks good, thanks for this PR!

File: e2e-tests/js_scripts/0006-proof_should_pay.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+const MIN_PRICE = 1000000000;
```
Comment (JackPiri) [inline e2e-tests/js_scripts/0006-proof_should_pay.js:9]: What's the rationale behind value of `MIN_PRICE`? I would have set this to `1 `(or just left the assertion something like `balanceBefore<balanceAfter`), but maybe there is something I'm missing.

Comment (la10736) [reply e2e-tests/js_scripts/0006-proof_should_pay.js:9]: I just used something that isn't trivial but sensible lesser than the one on the weight. I don't want both to strongly couple it to the used weight and to use a trivial value that is comparable to an nop extrinsic.

Comment (JackPiri) [reply e2e-tests/js_scripts/0006-proof_should_pay.js:9]: Ok thanks

**Pull request merged**